lib/Tool/Frontend.php

```diff
@@ -51,25 +52,49 @@ public static function isDocumentInCurrentSite(Document $document): bool
 
     public static function getSiteForDocument(Document $document): ?Site
     {
-        $cacheKey = 'sites_full_list';
-        if (RuntimeCache::isRegistered($cacheKey)) {
-            $sites = RuntimeCache::get($cacheKey);
-        } else {
-            $sites = new Site\Listing();
-            $sites->setOrderKey('(SELECT LENGTH(`path`) FROM documents WHERE documents.id = sites.rootId) DESC', false);
-            $sites = $sites->load();
-            RuntimeCache::set($cacheKey, $sites);
+        $siteIdOfDocument = self::getSiteIdForDocument($document);
+
+        if(!$siteIdOfDocument) {
+            return null;
         }
 
-        foreach ($sites as $site) {
-            if (strpos($document->getRealFullPath(), $site->getRootPath() . '/') === 0 || $site->getRootDocument()->getId() == $document->getId()) {
-                return $site;
+        return Site::getById($siteIdOfDocument);
+
+    }
+
+    public static function getSiteIdForDocument(Document $document): ?int
+    {
+        $siteMapping = self::getSiteMapping();
+
+        foreach ($siteMapping as $sitePath => $id) {
+            if (str_starts_with($document->getRealFullPath(), $sitePath)) {
+                return $id;
             }
         }
 
         return null;
     }
 
+    private static function getSiteMapping() : array
```

Comment:
@martineiber can you please check how often this method get's called on a normal document request? If it's called more than once, we should implement `RuntimeCache` in addition to the normal cache as well.

Reply:
@brusch If a document with in a site is called, then the method will be called more than once. 
I've added the `RuntimeCache` as well.